frontend/src/app/shared/components/calendar/calendar.component.ts

```diff
@@ -1,14 +1,57 @@
-import { Component, Input } from '@angular/core';
+import { Component, forwardRef, Input } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
 
 @Component({
     selector: 'app-calendar',
     templateUrl: './calendar.component.html',
     styleUrls: ['./calendar.component.sass'],
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            // eslint-disable-next-line no-use-before-define
+            useExisting: forwardRef(() => CalendarComponent),
+            multi: true,
+        },
+    ],
 })
-export class CalendarComponent {
+
+export class CalendarComponent implements ControlValueAccessor {
+    @Input() placeholder: string;
+
     maxDate = new Date();
 
     minDate = new Date(1900, 0, 1);
 
-    @Input() placeholder: string;
+    _value: Date = new Date();
+
+    // eslint-disable-next-line no-unused-vars,no-empty-function
+    onChange = (_: unknown) => {};
+
+    // eslint-disable-next-line no-unused-vars,no-empty-function
+    onTouch = (_: unknown) => {};
```

Comment:
Do you need this?
If yes - remove eslint-disable

Reply:
and remove unknown everywhere